protocol/str.go

```diff
@@ -0,0 +1,19 @@
+package protocol
+
+import "github.com/coniks-sys/coniks-go/merkletree"
+
+type DirSTR struct {
+	merkletree.SignedTreeRoot
```

Comment:
1c8418de seems to work.

But, I'm not sure I follow what you're saying.  Are `DirSTR`s retained?  They'll just be collected, no?  It seems like that just saves some copies.